### PR TITLE
`Resource<A, E>` and `ScopedRed<A>` is subtype of `Effect`.

### PR DESCRIPTION
**Pull request by KhraksMamtsov**
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

`Resource<A, E>` is subtype of `Effect<A, E>`.
`ScopedRed<A>` is subtype of `Effect<A>`.

## Related

- Related Issue #
- Closes #

### CODE REVIEW

File: packages/effect/src/internal/resource.ts [hunk 1]
```diff
@@ -46,11 +47,18 @@ export const manual = <A, E, R>(
   core.flatMap(core.context<R>(), (env) =>
     pipe(
       scopedRef.fromAcquire(core.exit(acquire)),
-      core.map((ref) => ({
-        [ResourceTypeId]: resourceVariance,
-        scopedRef: ref,
-        acquire: core.provideContext(acquire, env)
-      }))
+      core.map((ref) => {
+        const resource = {
```
Comment (tim-smart) [inline packages/effect/src/internal/resource.ts:51]: I think we can move the prototype out of here and use `Object.create`.

File: packages/effect/src/internal/scopedRef.ts [hunk 2]
```diff
@@ -41,11 +41,12 @@ export const fromAcquire = <A, E, R>(
         core.flatMap((value) =>
           circular.makeSynchronized([newScope, value] as const).pipe(
             core.flatMap((ref) => {
-              const scopedRef: ScopedRef.ScopedRef<A> = {
-                [ScopedRefTypeId]: scopedRefVariance,
-                pipe() {
-                  return pipeArguments(this, arguments)
+              const scopedRef = {
```
Comment (tim-smart) [inline packages/effect/src/internal/scopedRef.ts:44]: Same as resource, move the prototype out so it is shared by all instances.

File: packages/effect/src/internal/resource.ts (hunk 1 above)
Comment (KhraksMamtsov) [reply packages/effect/src/internal/resource.ts]: sure

File: packages/effect/src/internal/scopedRef.ts (hunk 2 above)
Comment (KhraksMamtsov) [reply packages/effect/src/internal/scopedRef.ts]: done

**Review by tim-smart — APPROVED** [review]

**Pull request merged**